### PR TITLE
Exogw 358: Enable Typescript on Examples and End-to-end

### PR DESCRIPTION
**Pull request by steve-keep**
## Summary by CodeRabbit

- **New Features**
  - Added `primaryKeyField` to various admin UI metadata types for better key management.
  - Introduced TypeScript type checking in the build process.

- **Bug Fixes**
  - Corrected port parsing for PostgreSQL and MySQL connections.

- **Refactor**
  - Removed inheritance from `BaseEntity` in several database entity classes.
  - Updated relationship field handling in various schema files.

- **Dependencies**
  - Added `vite` and `vite-plugin-svgr` to development dependencies.
  - Updated TypeScript to version 5.4.5.

- **Chores**
  - Updated `tsconfig.json` across multiple examples to include `skipLibCheck`.

### CODE REVIEW

**Comment by steve-keep** [comment]
@coderabbitai review

File: src/examples/databases/tsconfig.json [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+		"skipLibCheck": true
 	},
-	"include": ["./src"],
-	"references": [
-		{ "path": "../../packages/core" },
-		{ "path": "../../packages/mikroorm" },
-		{ "path": "../../packages/scalars" }
-	]
+	"exclude": ["**/node_modules/**", "**/lib/**"],
+	"include": ["./src"]
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/examples/databases/tsconfig.json:15]: Not required as inheritied

File: src/examples/aws-cognito/tsconfig.json [hunk 2]
```diff
@@ -9,11 +9,8 @@
...
-		"jsx": "react-jsx",
-		"types": ["vite/client", "vite-plugin-svgr/client"],
-		"sourceMap": true
+		"skipLibCheck": true
 	},
-	"include": ["./src"],
-	"references": [{ "path": "../../packages/core" }],
-	"sourceMap": true
+	"exclude": ["**/node_modules/**", "**/lib/**"],
+	"include": ["./src"]
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/examples/aws-cognito/tsconfig.json:15]: Not required

File: src/examples/sqlite/tsconfig.json [hunk 3]
```diff
@@ -8,7 +8,8 @@
 		"target": "es2019",
 		"module": "CommonJS",
 		"allowSyntheticDefaultImports": true,
-		"esModuleInterop": true
+		"esModuleInterop": true,
+		"skipLibCheck": true
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/examples/sqlite/tsconfig.json:12]: `skipLibCheck` is inherited from the root config.

File: src/examples/databases/src/backend/database.ts [hunk 4]
```diff
@@ -11,7 +11,7 @@
 		dbName: process.env.POSTGRES_DB_NAME ?? 'todo_app',
 		user: process.env.POSTGRES_DB_USER ?? 'postgres',
 		password: process.env.POSTGRES_DB_PASSWORD ?? '',
-		port: process.env.POSTGRES_DB_PORT ?? 5432,
+		port: parseInt(process.env.POSTGRES_DB_PORT ?? '5432'),
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/examples/databases/src/backend/database.ts:14]: erm...

File: src/examples/databases/src/backend/entities/postgresql/user.ts [hunk 5]
```diff
@@ -1,8 +1,7 @@
 import { BigIntType, Entity, PrimaryKey, Property } from '@mikro-orm/core';
-import { BaseEntity } from '@exogee/graphweaver-mikroorm';
 
 @Entity()
-export class User extends BaseEntity {
+export class User {
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/examples/databases/src/backend/entities/postgresql/user.ts:4]: These all use the rest entity

File: .github/workflows/windows-end-to-end-tests.yml [hunk 6]
```diff
@@ -37,7 +37,7 @@ jobs:
         run: pnpm i
 
       - name: Build CLI tool
-        run: pnpm build
+        run: pnpm build:packages
```
Comment (steve-keep) [inline .github/workflows/windows-end-to-end-tests.yml:40]: So it seem as if Xero does not build on Windows, so using a packages only build script.

File: src/examples/rest/src/backend/schema/task.ts [hunk 7]
```diff
@@ -93,7 +93,7 @@ export class Task {
 	@Field(() => Boolean)
 	isCompleted!: boolean;
 
-	@RelationshipField<OrmTask>(() => User, { id: 'userId' })
+	@RelationshipField<OrmTask>(() => User, { id: (entity) => entity.userId })
```
Comment (thekevinbrown) [inline src/examples/rest/src/backend/schema/task.ts:96]: Interesting, this should work with the `'userId'`, no? If not, we probably want to fix that up.

**Review by thekevinbrown — APPROVED** [review]

File: src/examples/rest/src/backend/schema/task.ts (hunk 7 above)
Comment (steve-keep) [reply src/examples/rest/src/backend/schema/task.ts:96]: Cool I was not sure if that was by design, It currently fails with the error:

`Cannot return null for non-nullable field Task.user.`

Comment (thekevinbrown) [reply src/examples/rest/src/backend/schema/task.ts:96]: Right, let's go ahead and merge this and create a backlog task to restore the string behaviour for simple field things.

Comment (steve-keep) [reply src/examples/rest/src/backend/schema/task.ts]: it is restored now as part of this PR

File: src/packages/core/src/decorators/relationship-field.ts [hunk 8]
```diff
@@ -3,7 +3,7 @@ import { graphweaverMetadata } from '../metadata';
 
 type RelationshipFieldOptions<D> = {
 	relatedField?: keyof D & string;
-	id?: (keyof D & (string | number)) | ((dataEntity: D) => string | number | undefined);
+	id?: (keyof D & string) | ((dataEntity: D) => string | number | undefined);
```
[the remark this answers is not in the document]
Comment (thekevinbrown) [reply src/packages/core/src/decorators/relationship-field.ts:6]: Are you saying that `undefined` is the same as `any`? It's not....

**Pull request merged**